### ISSUE TITLE
'harakiri' timeout is effectively a multiple of the one specified in case post-buffering is enabled and request's body is non-empty

### ISSUE THREAD

**Issue by andrii-rubtsov** [unbit/uwsgi#2030]
TLDR:
If:
1) `post-buffering` is ON (otherwise uWSGI emits warning)
2) your request (GET, POST, whatever) contains non-empty body
...then effective _harakiri_ timeout will be a multiple of that specified on startup. This is both unexpected (since not documented unfortunately) and unreliable, thus may not be usable for practical purposes.

Based on our debugging, the feature currently works like this.
uWGI's reads entire request body before sending it to the app. It reads body data in chunks. What is the most interesting and unexpected (since not documented) is that for every chunk effective harakiri timeout is added to the configured one: https://github.com/unbit/uwsgi/blob/7e8e113921076a364a521db75e8f91d3bedb5fdb/core/reader.c#L480

# Workaround
Use `user harakiri` instead.
